controller/index.js

```diff
@@ -1,1019 +1,1054 @@
+            }
+            return res.send(itemJson)
           }
-          // get the esri json data for the service
-          _get(req.params.id, req.params.item, key, req.query, function( err, itemJson ){
-              // when silent is sent as a param undefined
-              if (typeof req.params.silent === 'undefined') {
-                if (err) {
-                  if ( err.code && err.error ){
-                    res.status(err.code).send( err.error );
-                  } else {
-                    res.status(404).send( err );
-                  }
-                } else {
-                  if ( itemJson && itemJson.data && itemJson.data[0].features.length > 1000){
-                    itemJson.data[0].features = itemJson.data[0].features.splice(0,1000);
-                  }
-                  res.send( itemJson );
-                }
-              }
-          });
-        }
+        })
       }
-    });
-  };
+    })
+  }
 
-  controller.requestNewFile = function( req, res, dir, key, err, itemJson ){
-    if (err){
-      if ( err.code && err.error ){
-        res.status(err.code).send( err.error );
-      } else {
-        res.status(400).send( err );
-      }
-    } else if ( itemJson && itemJson.data && itemJson.data[0] && (!itemJson.data[0].features || !itemJson.data[0].features.length)) {
-      agol.log('error', req.url +' No features in data');
-      res.status(404).send( 'No features exist for the requested FeatureService layer');
-    } else {
+  /**
+   * Create key used to query a table in the cache
+   *
+   * @params {string} type - the type of the providers ("agol")
+   * @params {object} params - an object with an item id and layer number
+   *
+   * @returns {string} key
+   * @private
+   */
+  controller._createTableKey = function (type, params) {
+    return [type, params.item, (params.layer || 0)].join(':')
+  }
 
-      var name = ( itemJson && itemJson.data && itemJson.data[0] && (itemJson.data[0].name || (itemJson.data[0].info && itemJson.data[0].info.name) ) ) ? itemJson.data[0].name || itemJson.data[0].info.name : itemJson.name || itemJson.title;
-      // cleanze the name
-      name = name.replace(/\/|,|&\|/g, '').replace(/ /g, '_').replace(/\(|\)|\$/g, '');
-      name = (name.length > 150) ? name.substr(0, 150): name;
-
-      var fileParams = {
-        dir: dir,
-        key: key, 
-        id: req.params.item,
-        format: req.params.format,
-        type: 'agol',
-        data: (itemJson && itemJson.data && itemJson.data[0]) ? itemJson.data[0] : null
+  /**
+   * Respond to a request for file downloads when a dataset is still "processing"
+   * if the file exists send it, else return processing 202
+   *
+   * @params {object} type - the type of the providers ("agol")
+   * @params {object} params - an object with an item id and layer number
+   * @params {object} info - item metadata from the cache
+   * @private
+   */
+  controller._returnProcessingFile = function (req, res, info) {
```

Comment:
IMO this method should just be getting the processing state and returning that, it seems very strange to create a file path or return a actual file from this method